### DISCUSSION TITLE
Custom filesystem to use in python

### DISCUSSION THREAD

**Discussion by bajrangCoder** [pyodide/pyodide#4320] [Q&A]
How can I add my custom fs wrapper for interacting with filesystem from browser?

So user can do all fs related stuff from their python code.

Here is my fs wrapper: https://acode.app/plugin-docs/fs-operation?title=Fs%20Operation

**Comment by ryanking13** [comment]
Pyodide depends on Emscripten, and you need to implement a file system that is compatible with Emscripten.

Unfortunately, emscripten does not have good documentation for writing a custom file system (https://github.com/emscripten-core/emscripten/issues/9372), so you'll need to check file system implementations from Emscripten codes ([1](https://github.com/emscripten-core/emscripten/blob/main/src/library_idbfs.js), [2](https://github.com/emscripten-core/emscripten/blob/main/src/library_memfs.js), [3](https://github.com/emscripten-core/emscripten/blob/main/src/library_nodefs.js)).

After implementing a file system, you can mount it to the Pyodide file system using [`pyodide.FS.mount` method](https://emscripten.org/docs/api_reference/Filesystem-API.html#FS.mount).
